File: cw/cpp/two_sum.cpp
// https://www.codewars.com/kata/52c31f8e6605bcc646000082/train/cpp


#include <iostream>
#include <vector>
#include <utility>

std::pair<std::size_t, std::size_t> two_sum(const std::vector<int>& numbers, int target) {
    std::vector<int> index;

    for(int i = 0; i < numbers.size(); ++i) {
        for(int j = 0; j < i; ++j) {
            if((numbers[i] + numbers[j]) == target) {
                index.push_back(j);
                index.push_back(i);
                break;
            }
        }
    }

    std::pair<std::size_t, std::size_t> sum;
    sum.first = index[0];
    sum.second = index[1];

    return sum;
}

int main() {
    int lim;

    std::cin >> lim;
    int vals;
    std::vector<int> numbers;
    for(int i = 0; i < lim; ++i) {
        std::cin >> vals;
        numbers.push_back(vals);
    }
    int target;
    std::cin >> target;

    std::pair<std::size_t, std::size_t> sum = two_sum(numbers, target);
    std::cout << sum.first << ", " << sum.second << "\n";

    return 0;
}

File: cw/java/TwoSum.java
// https://www.codewars.com/kata/52c31f8e6605bcc646000082/train/java


import java.util.Scanner;

public class TwoSum {
    public static int[] twoSum(int[] numbers, int target) {
        int[] index = new int[2];

        for(int i = 0; i < numbers.length; ++i) {
            for(int j = 0; j < i; ++j) {
                if((numbers[i] + numbers[j]) == target) {
                    index[0] = j;
                    index[1] = i;
                }
            }
        }

        return index;
    }

    public static void main(String[] arse) {
        Scanner scan = new Scanner(System.in);
        
        System.out.print("Input array size: ");
        int lim = scan.nextInt();
        int[] numbers = new int[lim];
        
        for(int i = 0; i < lim; ++i) {
            System.out.print("Input: ");
            numbers[i] = scan.nextInt();
        }
        
        System.out.print("Input target: ");
        int target = scan.nextInt();

        scan.close();
        int[] output = twoSum(numbers, target);
        for(int i : output) {
            System.out.print("i: " + i);
        }
    }
}